src/inmanta/protocol/methods_v2.py

```diff
@@ -521,6 +521,7 @@ def get_resource_actions(
     action_id: Optional[uuid.UUID] = None,
     first_timestamp: Optional[datetime.datetime] = None,
     last_timestamp: Optional[datetime.datetime] = None,
+    exclude_changes: Optional[list[str]] = None,
```

Comment:
I had another failure in the `process` function of `rest/__init__.py:`

`isinstance(self._message[arg], typing_inspect.get_args(arg_type)[0])` is false whenever an enum is used as it becomes:

`isinstance('nochange', Change)` for example.

I thought it was ok to do it like this

Reply:
There are two main reasons why we shouldn't use type `str`:

* First of all we need to add the code that converts the `str` type into the `Change` type to the `get_resource_actions()` method and over time this code will be copied to more and more API endpoints. So having this type conversion/validation once in the protocol layer is much better. This also leads to error messages that are more consistent across API endpoints.
* Secondly, the signature of these methods is used to generate the swagger documentation. If we use type `str` it will be unclear for the user which values are allowed. We could solve that by adding the possible values to the docstring, but experience learns us that the docstring easily get out-of-sync with the code. If we use the enum type in the signature, the possible values are kept up-to-date automatically based on the implementation of the enum.

Reply:
I will try to add to fix this but I don't know how yet. I'll look tomorrow.